### DISCUSSION TITLE
PocketBase: Some pages requiring hard refresh

### DISCUSSION THREAD

**Discussion by adamshand** [caprover/caprover#1842] [General]
Hello,

I've installed PocketBase (v0.16.8) via the one-click installer.  There are no errors during or after the installation and everything looks as expected.  I've also tried with the latest (0.17.4) version and get the same behaviour.

EXCEPT that when I click links within PocketBase the new page doesn't load.  The URL changes, but the page content doesn't.  I get this error in the browser console:

`Uncaught (in promise) TypeError: Cannot read properties of null (reading 'insertBefore')`

Force reloading the page loads the page as expected and it seems to work.

I don't think (?) this is a PocketBase issue, I'm running the exact same version of PocketBase on another server (without Caprover) and PB behaves as expected.

I configured a mapped port in Caprover and see the same behaviour when accessing PocketBase via http and the mapped port (which I presume is bypassing Ngnix?).

So I'm not quite sure where to look for the problem.  I'm new to Caprover, any pointers would be appreciated.

Thanks,
Adam.

**Comment by githubsaturn** [comment]
When you say outside of CapRover is working fine, do you have it run it in a container? If not, try this dockerfile and see if it works:
```
FROM alpine:3.16.2
RUN apk add --no-cache unzip openssh
ADD https://github.com/pocketbase/pocketbase/releases/download/v0.16.3/pocketbase_0.16.3_linux_amd64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb/
EXPOSE 8080
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080"]
```

**Reply by githubsaturn** [reply]
Also I just tried installing it on a CapRover server and it seems to work fine:
- http://pocketbase.server.demo.caprover.com/_/
- `noreply@gmail.com / 123456123456`

**Reply by adamshand** [reply]
Thanks for the help!  Interesting, your version works fine for me.

I do have it running in a container, but a different image: https://hub.docker.com/r/spectado/pocketbase/

I'll try your Dockerfile and then try it with the same image I'm using elsewhere. 

I assume what's happening is that for some reason the frontend can't reach the Go backend ... but not quite sure what wold cause that.

**Reply by adamshand** [reply]
Hrm, I get exactly the same thing with your Dockerfile.  The deployment seems to work fine, I can login but clicking the Settings icon change the URL but doesn't load the settings page.  Shift reload and page content loads as expected.

There's nothing helpful in the logs:

❯ docker service logs srv-captain--pb.1.fd2xl2rj3mtv2l65ep1ll7kxb --since 60m --follow
srv-captain--pb.1.fd2xl2rj3mtv@syd1    | 2023/08/18 09:29:10 Server started at http://0.0.0.0:8080
srv-captain--pb.1.fd2xl2rj3mtv@syd1    |  ➜ REST API: http://0.0.0.0:8080/api/
srv-captain--pb.1.fd2xl2rj3mtv@syd1    |  ➜ Admin UI: http://0.0.0.0:8080/_/
^C

**Reply by adamshand** [reply]
Seems like it must be something I've done wrong with the Caprover installation, but ... 

I have two SvelteKit apps deployed which seem to be working fine, and I just installed Vaultwarden via the one-click apps and it's working fine.

**Reply by adamshand** [reply]
And I just tried installing the container directly from:  https://hub.docker.com/r/spectado/pocketbase/

Same thing.  So it's clearly something about my Caprover setup. Unless anyone has any suggestions I guess I'm tearing it all down and building it again.  Hopefully I just did something stupid, but I can't see anything suggestions about what it might be in the logs ...

**Reply by adamshand** [reply]
<sigh> sorry for the noise.  Turns out that this was a badly behaving browser.  Restarted Brave and everything came right.

**Reply by githubsaturn** [reply]
hah! interesting! good find!
